storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6044,35 +6158,53 @@ int ha_rocksdb::read_after_key(const Rdb_key_def &kd,
     with HA_READ_KEY_OR_NEXT, $GT = '>='
   */
   rocksdb_smart_seek(kd.m_is_reverse_cf, m_scan_it, key_slice);
-  return m_scan_it->Valid() ? HA_EXIT_SUCCESS : HA_ERR_KEY_NOT_FOUND;
+
+  for (;;) {
+    if (!m_scan_it->Valid()) {
+      return HA_ERR_KEY_NOT_FOUND;
+    }
+
+    /*
+      If TTL is enabled we need to check if the given key has already expired
+      from the POV of the current transaction.  If it has, try going to the next
+      key.
+    */
+    if (kd.has_ttl() &&
+        should_hide_ttl_rec(m_scan_it->value(), tx->m_snapshot_timestamp)) {
+      rocksdb_smart_next(kd.m_is_reverse_cf, m_scan_it);
+      continue;
+    }
+
+    break;
+  }
+
+  return HA_EXIT_SUCCESS;
 }
 
-int ha_rocksdb::position_to_correct_key(const Rdb_key_def &kd,
-                                        const enum ha_rkey_function &find_flag,
-                                        const bool &full_key_match,
-                                        const uchar *const key,
-                                        const key_part_map &keypart_map,
-                                        const rocksdb::Slice &key_slice,
-                                        bool *const move_forward) {
+int ha_rocksdb::position_to_correct_key(
+    const Rdb_key_def &kd, Rdb_transaction *const tx,
```

Comment:
Instead of passing in the Rdb_transaction, can we pass in the filter time to be explicit in how the transaction is being used?